File: src/main/java/ca/jonathanfritz/funopoly/Dice.java
package ca.jonathanfritz.funopoly;

import java.util.Random;

public class Dice {
	private final Random random;

	public Dice() {
		random = new Random();
	}

	public Dice(long seed) {
		random = new Random(seed);
	}

	public DiceRollResult roll() {
		final int die1 = random.nextInt(6) + 1;
		final int die2 = random.nextInt(6) + 1;
		return new DiceRollResult(die1 == die2, die1 + die2);
	}

	public class DiceRollResult {
		public final boolean isDoubles;
		public final int value;

		public DiceRollResult(boolean isDoubles, int value) {
			this.isDoubles = isDoubles;
			this.value = value;
		}
	}
}

File: src/main/kotlin/ca/jonathanfritz/funopoly/game/Dice.kt
package ca.jonathanfritz.funopoly.game

import kotlin.random.Random

class Dice(private val seed: Long = System.currentTimeMillis()) {

    private val random: Random = Random(seed)

    /**
     * Rolls the dice - returns a pair containing the sum of the two dice in the first value,
     * and a boolean indicating if doubles were rolled in the second
     */
    fun roll(): Pair<Int, Boolean> {
        val first = random.nextInt(1, 7)
        val second = random.nextInt(1, 7)
        return Pair(first + second, first == second)
    }
}